core/src/main/scala/hydra/core/akka/SchemaRegistryActor.scala

```diff
@@ -0,0 +1,196 @@
+package hydra.core.akka
+
+import akka.actor.{Actor, Props}
+import akka.cluster.pubsub.DistributedPubSub
+import akka.cluster.pubsub.DistributedPubSubMediator.{Publish, Subscribe}
+import akka.pattern.{CircuitBreaker, pipe}
+import com.typesafe.config.Config
+import hydra.avro.registry.{ConfluentSchemaRegistry, SchemaRegistryException}
+import hydra.avro.resource.{SchemaResource, SchemaResourceLoader}
+import hydra.common.logging.LoggingAdapter
+import hydra.core.protocol.HydraApplicationError
+import org.apache.avro.{Schema, SchemaParseException}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+/**
+  * This actor serves as an proxy between the handler registry
+  * and the application.
+  *
+  * Created by alexsilva on 12/5/16.
+  */
+class SchemaRegistryActor(config: Config, settings: Option[CircuitBreakerSettings]) extends Actor
+  with LoggingAdapter {
+
+  import SchemaRegistryActor._
+  import context.dispatcher
+
+  val breakerSettings = settings getOrElse new CircuitBreakerSettings(config)
+
+  private val registryFailure: Try[SchemaRegistryResponse] => Boolean = {
+    case Success(_) => false
+    case Failure(ex) if ex.isInstanceOf[SchemaRegistryException] => false
+    case Failure(_) => true
+  }
+
+  private val breaker = CircuitBreaker(
+    context.system.scheduler,
+    maxFailures = breakerSettings.maxFailures,
+    callTimeout = breakerSettings.callTimeout,
+    resetTimeout = breakerSettings.resetTimeout)
+    .onOpen(notifyOnOpen())
+
+  val registry = ConfluentSchemaRegistry.forConfig(config)
+
+  val loader = new SchemaResourceLoader(registry.registryUrl, registry.registryClient)
+  val mediator = DistributedPubSub(context.system).mediator
+  mediator ! Subscribe(SchemaRegisteredTopic, self)
+
+  def getSubject(schema: Schema): String = addSchemaSuffix(schema.getFullName)
+
+  override def receive = {
+    case FetchSchemaRequest(location) =>
+      val futureResource = loader.retrieveSchema(location).map(resource => FetchSchemaResponse(resource))
+      breaker.withCircuitBreaker(futureResource, registryFailure) pipeTo sender
+
+    case RegisterSchemaRequest(json: String) =>
+      val tryRegisterSchema = tryHandleRegisterSchema(json)
```

Comment:
We should do the Future here; this way this will fail if schema registry is down.

val tryRegisterSchema = Future.fromTry(tryRegisterSchema)